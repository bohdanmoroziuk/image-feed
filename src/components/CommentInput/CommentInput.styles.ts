import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderBottomColor: 'rgba(0, 0, 0, 0.1)',
    paddingHorizontal: 20,
    height: 60,
  },
  input: {
    flex: 1,
  },
});
