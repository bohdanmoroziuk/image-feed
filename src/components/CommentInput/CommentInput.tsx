import { FunctionComponent, useState } from 'react';
import { TextInput, View } from 'react-native';

import styles from './CommentInput.styles';

export interface CommentInputProps {
  placeholder?: string;
  onSubmit: (text: string) => void;
}

export const CommentInput: FunctionComponent<CommentInputProps> = ({
  placeholder = '',
  onSubmit,
}) => {
  const [text, setText] = useState('');

  const handleChangeText = (text: string) => {
    setText(text);
  };

  const handleSubmitEditing = () => {
    if (!text) return;

    onSubmit(text);
    setText('');
  };

  return (
    <View style={styles.container}>
      <TextInput
        style={styles.input}
        value={text}
        placeholder={placeholder}
        underlineColorAndroid="transparent"
        onChangeText={handleChangeText}
        onSubmitEditing={handleSubmitEditing}
      />
    </View>
  );
};
