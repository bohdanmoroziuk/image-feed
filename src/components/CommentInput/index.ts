import { CommentInput } from './CommentInput';

export type { CommentInputProps } from './CommentInput';
export default CommentInput;
