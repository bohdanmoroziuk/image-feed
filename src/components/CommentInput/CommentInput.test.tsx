import { render, fireEvent } from '@testing-library/react-native';

import { CommentInput } from './CommentInput';

const mockOnSubmit = jest.fn();

afterEach(() => {
  jest.clearAllMocks();
});

describe('CommentInput', () => {
  it('renders successfully', () => {
    const { toJSON } = render(
      <CommentInput
        placeholder="Leave a comment"
        onSubmit={mockOnSubmit}
      />
    );

    expect(toJSON()).toMatchSnapshot()
  });

  it('calls the onSubmit handler', async () => {
    const { getByPlaceholderText } = render(
      <CommentInput
        placeholder="Leave a comment"
        onSubmit={mockOnSubmit}
      />
    );

    const input = getByPlaceholderText('Leave a comment');

    fireEvent.changeText(input, 'My very useful comment');

    fireEvent(input, 'submitEditing');

    expect(mockOnSubmit).toHaveBeenCalledWith('My very useful comment');
  });

  it('does not call the onSubmit handler when the text is empty', async () => {
    const { getByPlaceholderText } = render(
      <CommentInput
        placeholder="Leave a comment"
        onSubmit={mockOnSubmit}
      />
    );

    const input = getByPlaceholderText('Leave a comment');

    fireEvent(input, 'submitEditing');

    expect(mockOnSubmit).not.toHaveBeenCalled();
  });
});