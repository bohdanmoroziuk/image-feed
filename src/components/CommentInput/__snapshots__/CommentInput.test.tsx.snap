// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentInput renders successfully 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "rgba(0, 0, 0, 0.1)",
      "borderBottomWidth": 0.5,
      "height": 60,
      "paddingHorizontal": 20,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Leave a comment"
    style={
      Object {
        "flex": 1,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
